### ISSUE TITLE
Replace ExternalProcess by a simple Unzip step

### ISSUE THREAD

**Issue by mauricioaniche** [SERG-Delft/andy#165]
We don't use `ExternalProcess` as it was initially meant. We should simplify the code by getting rid of it. This is only used to unzip Selenium assignments, so we may need an extra unzip step.

**Comment by NeilAlwani** [comment]
I would like to work on this.

**Comment by mauricioaniche** [comment]
@martinmladenov any hints for this one?

**Comment by martinmladenov** [comment]
@NeilAlwani The original idea behind external processes was to have a Spring application run in the background for Selenium assignments. We were planning to start the application before running the student's tests, and for meta tests we were planning to have separate .jar files which we run instead. However, this proved to be too resource-intensive to run in an exam environment, so we scrapped the idea.

For Selenium assignments, we still use external processes, but in a very different way. In each Selenium assignment, we have something like this in the configuration:

```java
    @Override
    public ExternalProcess externalProcess() {
        return new CommandExternalProcess(
                "sh /user_code/extract.sh",
                "done extracting");
    }
```

This extract.sh script simply unzips the assignment and its meta tests. This was done as a workaround when we realised our external process idea wouldn't work.

We no longer need external processes, we only need an unzipping step. It should accept a list of .zip file names as strings and it should unzip them in Andy's working directory. All external process code (incl. external process meta tests) should then be removed.

**Comment by Arraying** [comment]
FWIW, I don't think an external shell process is the way to go with unzipping. Perhaps we should consider storing the list of files to extract somewhere (such as the Configuration), and then using something like `java.util.zip` to perform extraction. 

Right now, you must ensure that:
- `unzip` is installed (in the Dockerfile)
- the operating system supports shell execution (especially in our format), think about students running it outside Docker
- the `extract.sh` contains everything expected

That's 3 different places to update and maintain. In my experience, shells are always flakey especially across different OS, hence my suggestion.

**Comment by martinmladenov** [comment]
@Arraying Yes, this is exactly the point of this issue. We currently use a shell process for unzipping, and this is definitely a bit sketchy, so we want to get rid of it and replace it with a native unzipping step.

**Comment by NeilAlwani** [comment]
Please correct me where I am wrong @martinmladenov. From what I understand so far the `DirectoryConfiguration `will be given a list of .zip files that will be unzipped in the `OrganizeSourceCodeStep ` using the `unzip ` method that will be in the utils module.

**Comment by mauricioaniche** [comment]
Remark: there's no need to put the zip code in the utils folder. This function is likely to only be used by this part of the code so it's best if the entire feature is together, eg, on the same package.

The utils package was meant to be a package for functions that are used everywhere. It's too easy to lose this purpose. Maybe we need a ticket to revisit all the classes there and where they are used :)

**Comment by martinmladenov** [comment]
@NeilAlwani The list of files to unzip should be configurable in the RunConfiguration. I'd suggest making a new step which handles the unzipping. This should happen after the OrganizeSourceCodeStep and after loading the configuration (before that, we don't yet know what we should unzip, as the configuration only gets loaded then).

**Comment by NeilAlwani** [comment]
@martinmladenov I am trying to figure out how to test the unzipStep that I have created so far. Could you give me some hints? I see that I need to use Result to make my assertions.

**Comment by martinmladenov** [comment]
You could perhaps write an integration test which unzips a zip and checks for the presence of the unzipped files. Or are you asking about something else? If so, could you push your changes to a branch and make a draft PR so that I can see your code?

**Comment by NeilAlwani** [comment]
Alright, will do!
